src/protocols/vnc/vnc.c

```diff
@@ -512,6 +516,14 @@ void* guac_vnc_client_thread(void* data) {
         if (wait_result < 0)
             guac_client_abort(client, GUAC_PROTOCOL_STATUS_UPSTREAM_ERROR, "Connection closed.");
 
+        if (server_runing_start - server_runing_end > GUAC_VNC_IDLE_TIMEOUT) {
+         rfb_client = guac_vnc_get_client(client);
+         server_runing_end = guac_timestamp_current();
+            if (!rfb_client) {
```

Comment:
Is this just intended to test whether the connection has already closed?

Reply:
Well this is how it is working. If the system is idle for like say 10s. After 10s the guacd using `guac_vnc_get_client()`. In that when we call the libvnc API. If the connection is already connected. It will return true i.e. Client connected. If the server doesn't respond it will try to create a new connection if it still doesn't respond it will generate a error message.
First I tried using the socket ping function but in that the issue is and it is a issue reported in LibVNC as well. And is generated from their sub library TCP socket library, as on some platforms it doesn't return a error message. As explained here [TCP connection Error](https://stackoverflow.com/questions/283375/detecting-tcp-client-disconnect). 
So, I tried it by sending multiple write requests and that didn't work like took more than 5 mins before a response was generated and in some cases more than 15 mins.